static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,215 @@
+            userVariantParams.campaignCode = buildCampaignCode(
+                offering,
+                campaignId,
+                userVariant.id
+            );
+        }
+
+        return userVariantParams;
+    }
+    return {};
+};
+
+/*
+ * Params for the banner are overlaid in this order, earliest taking precedence:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition
+ *  * Offering ('membership' or 'contributions')
+ *  * Default
+ *
+ * The 'offering' in use comes from either:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition (only one offering can be the default for a given Edition)
+ *
+ * Returns either 'null' if no banner is available for this edition,
+ * otherwise a populated params object that looks like this:
+ *
+ *  {
+ *    minArticles: 5, // how many articles should the user see before they get the engagement banner?
+ *    messageText: "..."
+ *    colourStrategy: // a function to determine what css class to use for the banner's colour
+ *    buttonCaption: "Become a Supporter"
+ *  }
+ *
+ */
+const deriveBannerParams = (location: string): Object | string => {
+    const defaultParams = membershipEngagementBannerUtils.defaultParams(
+        location
+    );
+    const userTest = getUserTest();
+    const campaignId = userTest ? userTest.campaignId : undefined;
+    const userVariant = getUserVariant(userTest);
+
+    if (userVariant && userVariant.blockEngagementBanner) {
+        return DO_NOT_RENDER_ENGAGEMENT_BANNER;
+    }
+
+    return Object.assign(
+        {},
+        defaultParams,
+        getUserVariantParams(userVariant, campaignId, defaultParams.offering)
+    );
+};
+
+// Used to send an interaction if the engagement banner is shown.
+const recordInteraction = (interaction: Object): void => {
+    if (interaction) {
+        const component = interaction.component;
+        const value = interaction.value;
+
+        if (component && value) {
+            ophan.record({
+                component,
+                value,
+            });
+        }
+    }
+};
+
+const paypalAndCreditCardImage =
+    (config.images &&
+        config.images.acquisitions &&
+        config.images.acquisitions['paypal-and-credit-card']) ||
+    '';
+
+const selectSequentiallyFrom = (array: Array<any>): any =>
+    array[local.get('gu.alreadyVisited') % array.length];
+
+const showBanner = (params: Object): void => {
+    if (params === DO_NOT_RENDER_ENGAGEMENT_BANNER || isBlocked()) {
+        return;
+    }
+
+    const colourClass = params.colourStrategy();
+
+    const messageText = Array.isArray(params.messageText)
+        ? selectSequentiallyFrom(params.messageText)
+        : params.messageText;
+
+    const urlParameters = {
+        REFPVID: params.pageviewId,
+        INTCMP: params.campaignCode,
+    };
+
+    const linkUrl = `${params.linkUrl}?${constructQuery(urlParameters)}`;
+
+    const renderedBanner = template(messageTemplate, {
```

Comment:
Could we use es6 template literals here and drop the lodash `templates` library?